kedro/config/omegaconf_config.py

```diff
@@ -431,7 +433,7 @@ def _check_duplicates(seen_files_to_keys: dict[Path, set[Any]]):
             raise ValueError(f"{dup_str}")
 
     @staticmethod
-    def _resolve_environment_variables(config: dict[str, Any]) -> None:
+    def _resolve_environment_variables(config: DictConfig | ListConfig) -> None:
```

Comment:
This may be a bit controversial, technically OmegaConf support both but Kedro only expect the former one.

Reply:
No that's a good point. I've changed it.